src/worktree.rs

```diff
@@ -0,0 +1,335 @@
+use crate::buf::Buf;
+use crate::reference::Reference;
+use crate::repo::Repository;
+use crate::util::{self, Binding};
+use crate::{raw, Error};
+use std::os::raw::c_int;
+use std::path::Path;
+use std::ptr;
+use std::str;
+use std::{marker, mem};
+
+/// An owned git worktree
+///
+/// This structure corresponds to a `git_worktree` in libgit2.
+//
+pub struct Worktree {
+    raw: *mut raw::git_worktree,
+}
+
+// It is the current belief that a `Worktree` can be sent among threads, or
+// even shared among threads in a mutex
+unsafe impl Send for Worktree {}
```

Comment:
Oh sorry I think I missed this earlier, but are you sure of this? Is this documented in libgit2?

Reply:
This is the best I could find:
> Unless otherwise specified, libgit2 objects cannot be safely accessed by multiple threads simultaneously.
> [...]
> Use an object from a single thread at a time. Most data structures do not guard against concurrent access themselves. This is because they are rarely used in isolation and it makes more sense to synchronize access via a larger lock or similar mechanism.

—https://github.com/libgit2/libgit2/blob/master/docs/threading.md

I think this can be read as they are Send but not Sync?

Reply:
The original version of that file was clearer on this point:
> You may safely use any libgit2 object from any thread, though there may be issues depending on the cryptographic libraries libgit2 or its dependencies link to (more on this later).

—https://github.com/libgit2/libgit2/commit/15bea02c04ebf3172d495f14219b30eba262eb6a#diff-bb59cfb2bcd0e444b33e25dc3f1e127e095d72379150cc82d408afa63b9903bf

Reply:
Yeah as @joshtriplett [pointed out below](https://github.com/rust-lang/git2-rs/pull/603#issuecomment-728262191) libgit2's policy isn't enough to dictate whether things are `Send` or `Sync` in Rust. You'll need to devle into the internals and audit what's happening to see if it's safe. I think I did that for `git_repository` a long time ago, but if this wants to stay then it should happen for `git_worktree` as well. It's always ok to drop this `unsafe impl` for now though and add it back later if it works.

Reply:
You're right. I didn't notice that. I'll remove it.